client/src/actions/__tests__/teaActions.test.js

```diff
@@ -0,0 +1,202 @@
+  teaTypes: ["Black", "Green", "White", "Herbal"]
+});
+
+const mockListOwner = "5cf18ae7d39d81638810de09";
+
+const mockDeleteTeaID = "25070e52-e635-4883-ae9b-583113573b9f";
+
+const mockAddTea = {
+  userID: "5cf18ae7d39d81638810de09",
+  teaID: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+  name: "Lapsang Souchang",
+  brand: "McNulty's",
+  teaType: "Black",
+  servings: "12",
+  id: "3d4a3605-4368-4ee1-abe4-318d7982e491"
+};
+
+const mockEditTea = {
+  flash: { name: "", teaID: "" },
+  touched: { name: false, servings: true },
+  userID: "5cf18ae7d39d81638810de09",
+  currentTea: {
+    id: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+    name: "Lapsang Souchang",
+    brand: "McNulty's",
+    teaType: "Black",
+    servings: 12
+  },
+  teaID: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+  name: "Lapsang Souchang",
+  brand: "McNulty's",
+  teaType: "Black",
+  servings: "14",
+  edit: true,
+  brands: [],
+  brandsDataList: [],
+  id: "3d4a3605-4368-4ee1-abe4-318d7982e491"
+};
+
+const mockPostResponse = {
+  type: "ADD_TEA",
+  payload: {
+    id: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+    name: "Lapsang Souchang",
+    brand: "McNulty's",
+    teaType: "Black",
+    servings: 12
+  }
+};
+
+const mockPutResponse = {
+  type: "EDIT_TEA",
+  payload: {
+    id: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+    name: "Lapsang Souchang",
+    brand: "McNulty's",
+    teaType: "Black",
+    servings: 14
+  }
+};
+
+const mockGetResponse = {
+  type: "GET_TEAS",
+  payload: [
+    {
+      id: "63f48407-a224-49a9-80b9-3257b383d8e8",
+      name: "Sleepytime",
+      brand: "Celestial Seasonings",
+      teaType: "Herbal",
+      servings: 12
+    },
+    {
+      id: "3d4a3605-4368-4ee1-abe4-318d7982e491",
+      name: "Lapsang Souchang",
+      brand: "McNulty's",
+      teaType: "Black",
+      servings: 12
+    }
+  ]
+};
+
+let addTea = teaActions.addTea;
+let editTea = teaActions.editTea;
+let deleteTea = teaActions.deleteTea;
+let getTeas = teaActions.getTeas;
+
+jest.mock("../../lib/api", () => ({
+  post: jest.fn(() => Promise.resolve(mockPostResponse)),
+  put: jest.fn(() => Promise.resolve(mockPutResponse)),
+  delete: jest.fn(() => Promise.resolve()),
+  get: jest.fn(() => Promise.resolve(mockGetResponse))
+}));
+
+describe("addTea", () => {
+  beforeEach(() => {
+    store.clearActions();
+  });
+  test("returns a function", () => {
+    expect(addTea(mockAddTea)).toBeInstanceOf(Function);
+  });
+  test("it calls 'post' on the API with the correct path and the tea data", () => {
+    store.dispatch(addTea(mockAddTea));
+    expect(API.post).toHaveBeenCalledWith("/teas/new-tea", mockAddTea);
+  });
+  test("it returns the ADD_TEA action type and payload", async () => {
+    await store.dispatch(addTea(mockAddTea));
+    expect(store.getActions()[0].payload).toEqual(mockPostResponse);
+    expect(store.getActions()[0].type).toEqual("ADD_TEA");
+  });
+});
+
+describe("editTea", () => {
+  beforeEach(() => {
+    store.clearActions();
```

Comment:
It's probably easier to just re-create the store in a `beforeEach` block at the top level of this file.